src/KnowbaseItem.php

```diff
@@ -739,6 +744,20 @@ public function prepareInputForUpdate($input)
         if (isset($input["name"]) && empty($input["name"])) {
             $input["name"] = __('New item');
         }
+
+        if (
+            (
+                isset($input['allow_access_using_token'])
+                && $input['allow_access_using_token']
+                && empty($this->fields['token'])
+            ) || (
+                isset($input['_regenerate_token'])
+                && $input['_regenerate_token'] == 'on'
```

Comment:
Probably better to check for a thruthy value as people updating the item from the API might expect to send something like `_regenerate_token => true`. 

```suggestion
                && $input['_regenerate_token']
```